File: Easy/registry/meta_registryistitutiesteriview/meta_registryistitutiesteriview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_registryistitutiesteriview
{
    public class Meta_registryistitutiesteriview : Meta_easydata 
	{
        public Meta_registryistitutiesteriview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "registryistitutiesteriview") {
				Name = "Istituti esteri";
			EditTypes.Add("istitutiesteri");
			ListingTypes.Add("istitutiesteri");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "istitutiesteri": {
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "registry_cf", "Codice fiscale", nPos++);
						DescribeAColumn(T, "registry_p_iva", "Partita iva", nPos++);
						DescribeAColumn(T, "registry_active", "attivo", nPos++);
						DescribeAColumn(T, "geo_city_title", "Città", nPos++);
						DescribeAColumn(T, "geo_nation_title", "Nazione", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_idreg", "Identificativo", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_name", "Name", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_city", "City", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_code", "Code", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_institutionalcode", "Institutional code", nPos++);
						DescribeAColumn(T, "registry_istitutiesteri_referencenumber", "Reference number", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "istitutiesteri": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/registry/meta_registryistitutiesteriview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_registryistitutiesteriview() {
        MetaData.apply(this, ["registryistitutiesteriview"]);
        this.name = 'meta_registryistitutiesteriview';
    }

    meta_registryistitutiesteriview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_registryistitutiesteriview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'istitutiesteri':
						this.describeAColumn(table, 'registry_istitutiesteri_idreg', 'Identificativo', null, 1000, null);
						this.describeAColumn(table, 'registry_istitutiesteri_name', 'Name', null, 2000, 1024);
						this.describeAColumn(table, 'title', 'Denominazione', null, 3000, 101);
						this.describeAColumn(table, 'registry_istitutiesteri_city', 'City', null, 3000, 255);
						this.describeAColumn(table, 'registry_cf', 'Codice fiscale', null, 4000, 16);
						this.describeAColumn(table, 'registry_istitutiesteri_code', 'Code', null, 4000, 50);
						this.describeAColumn(table, 'registry_p_iva', 'Partita iva', null, 5000, 15);
						this.describeAColumn(table, 'registry_istitutiesteri_institutionalcode', 'Institutional code', null, 5000, 50);
						this.describeAColumn(table, 'registry_active', 'attivo', null, 6000, null);
						this.describeAColumn(table, 'registry_istitutiesteri_referencenumber', 'Reference number', null, 6000, 50);
						this.describeAColumn(table, 'geo_city_title', 'Città', null, 7100, 65);
						this.describeAColumn(table, 'geo_nation_title', 'Nazione', null, 8300, 65);
//$objCalcFieldConfig_istitutiesteri$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "istitutiesteri": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('registryistitutiesteriview', new meta_registryistitutiesteriview('registryistitutiesteriview'));

	}());
